File: miscellaneous/ltcd_198.py
'''
LeetCode - problem 198
'''

    def rob(self, nums: List[int]) -> int:
        max_2_houses_ago = max_last_house = 0
	for house_value in nums:
            max_2_houses_ago, max_last_house = max_last_house, max(max_2_houses_ago + house_value, max_last_house)
            
        return max_last_house

File: miscellaneous/ltcd_198.cpp
//LeetCode - problem 198

int rob(vector<int>& nums) {
	int len=nums.size();
	if(0==len) 
		return 0;
	vector<int> bag;
	if(len>0)  
		bag.push_back(nums[0]);
	if(len>1) 
		bag.push_back(max(nums[0],nums[1]));
	for(int i=2;i<len;++i)
		bag.push_back(max(bag[i-1],nums[i]+bag[i-2]));
	return bag[len-1];
}
